crypto/asn1/a_int.c

```diff
@@ -289,7 +289,7 @@ static int asn1_get_int64(int64_t *pr, const unsigned char *b, size_t blen,
             ASN1err(ASN1_F_ASN1_GET_INT64, ASN1_R_TOO_SMALL);
             return 0;
         }
-        *pr = -(int64_t)r;
+        *pr = -(uint64_t)r;
```

Comment:
> only legitimate reason to disregard warning is that compiler gets it wrong

I don't think that is true for *all* warnings implemented in *all* compilers.  For example, clang with -Weverything includes warnings when __attribute__(padded) is used on structures that would not have padding applied for the current architecture (but potentially would have padding added on *different* architectures), as well as several other warnings that are not applicable everywhere and only informational at best.